From f40951fad65e4cb9dec8a44f54ebf90daa5530b4 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Wed, 19 Dec 2018 15:38:09 +0530
Subject: [PATCH 0492/1851] documentation: dt: vpss-scaler: Fix support for
 clock framework

commit 378492b143bf76e555654a324b83b1819cf51b2e from
https://github.com/Xilinx/linux-xlnx.git

Fixes the support for clock framework by adding the axi lite clock as a
required device tree property. Now scaler needs two clocks aclk_axis
and aclk_ctrl. The compatible string has also been changed.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 .../bindings/media/xilinx/xlnx,v-vpss-scaler.txt     | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/Documentation/devicetree/bindings/media/xilinx/xlnx,v-vpss-scaler.txt b/Documentation/devicetree/bindings/media/xilinx/xlnx,v-vpss-scaler.txt
index cc0616784bbc..148c727b5416 100644
--- a/Documentation/devicetree/bindings/media/xilinx/xlnx,v-vpss-scaler.txt
+++ b/Documentation/devicetree/bindings/media/xilinx/xlnx,v-vpss-scaler.txt
@@ -3,11 +3,16 @@
 
 Required properties:
 
-- compatible: Must be "xlnx,v-vpss-scaler".
+- compatible: Must be "xlnx,v-vpss-scaler-1.0".
+  The older string "xlnx,v-vpss-scaler" will be deprecated.
 
 - reg: Physical base address and length of the registers set for the device.
 
 - clocks: Reference to the AXI Streaming clock feeding the VPSS Scaler AP_CLK
+  and AXI4 Lite control interface clock.
+
+- clock-names: Must contain "aclk_axis" and "aclk_ctrl" in the same order as
+  clocks listed in clocks property.
 
 - xlnx,num-hori-taps, xlnx,num-vert-taps: The number of horizontal and vertical
   taps for scaling filter(range: 2,4,6,8,10,12).
@@ -44,9 +49,10 @@ Optional properties:
 Example:
 
 	scaler_1:scaler@a0000000 {
-		compatible = "xlnx,v-vpss-scaler";
+		compatible = "xlnx,v-vpss-scaler-1.0";
 		reg = <0x0 0xa0000000 0x0 0x40000>;
-		clocks = <&vid_stream_clk>;
+		clocks = <&vid_stream_clk>, <&misc_clk_2>;
+		clock-names = "aclk_axis", "aclk_ctrl";
 		xlnx,num-hori-taps = <8>;
 		xlnx,num-vert-taps = <8>;
 		xlnx,pix-per-clk = <2>;
-- 
2.31.1

